Library/Formula/cairomm.rb

```diff
@@ -17,21 +14,59 @@ class Cairomm < Formula
 
   deprecated_option "without-x" => "without-x11"
 
-  depends_on 'pkg-config' => :build
+  depends_on "pkg-config" => :build
   if build.cxx11?
-    depends_on 'libsigc++' => 'c++11'
+    depends_on "libsigc++" => "c++11"
   else
-    depends_on 'libsigc++'
+    depends_on "libsigc++"
   end
 
-  depends_on 'cairo'
-  depends_on 'libpng'
+  depends_on "cairo"
+  depends_on "libpng"
   depends_on :x11 => :recommended
 
   def install
     ENV.cxx11 if build.cxx11?
     system "./configure", "--disable-dependency-tracking",
                           "--prefix=#{prefix}"
-    system "make install"
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.cc").write <<-EOS.undent
+      #include <string>
+      #include <iostream>
+      #include <cmath>
+      #include <cairommconfig.h>
+      #include <cairomm/context.h>
+      #include <cairomm/surface.h>
+
+      int main() {
+        Cairo::RefPtr<Cairo::ImageSurface> surface =
+          Cairo::ImageSurface::create(Cairo::FORMAT_ARGB32, 600, 400);
+
+        Cairo::RefPtr<Cairo::Context> cr = Cairo::Context::create(surface);
+
+        cr->save();
+        cr->set_source_rgb(0.86, 0.85, 0.47);
+        cr->paint();
+
+        std::string filename = "image.png";
+        surface->write_to_png(filename);
+      }
+    EOS
+    flags = %W[
+      -I#{lib}/cairomm-1.0/include
+      -I#{include}/cairomm-1.0
+      -I#{Formula["cairo"].opt_include}/cairo
+      -I#{Formula["freetype"].opt_include}/freetype2
+      -I#{Formula["libsigc++"].opt_include}/sigc++-2.0
+      -I#{Formula["libsigc++"].opt_lib}/sigc++-2.0/include
+      -L#{lib}
+      -lcairomm-1.0
+    ]
+    system ENV.cxx, *flags, "test.cc", "-o", "test"
```

Comment:
This syntax doesn't work in Ruby 1.8: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/24249/version=mountain_lion/console

`*args` have to be at the end of the argument list. You can do `*(flags + ["some", "other", "things"])` but it's a little ugly; may as well just push everything into flags?